### DISCUSSION TITLE
Enormous number of API requests, using getStaticProps with revalidate

### DISCUSSION THREAD

**Discussion by howells** [vercel/next.js#17739] [Help]
I've seen a few other folk have this issue here. I've started using getStaticProps with revalidate, but for fairly low traffic sites I'm seeing *enormous* load on my API server. My app is on Vercel. Revalidation is set anywhere between 300 - 3600 seconds, but this doesn't seem to alleviate the issues, and I'm struggling to work out what is going on.

My getStaticProps typically look something like this:

```
export async function getStaticProps({ params }) {
  const uri = params.slug.join('/');

  const navigationResponse = await fetchAPI(fetchNavigation);
  const response = await fetchAPI(fetchPage, { uri });

  return {
    props: { page: response.page, uri, navigation: navigationResponse },
    revalidate: 300,
  };
}
```

Where fetchAPI is just a pretty standard fetch call:

```
const fetchAPI = async (query, variables = {}, token = null) => {
  const url = token
    ? `${process.env.NEXT_PUBLIC_API_URL}?token=${token}`
    : `${process.env.NEXT_PUBLIC_API_URL}`;

  try {
    const res = await fetch(url, {
      method: 'POST',
      headers: {
        'Content-Type': 'application/json',
        Authorization: `Bearer ${process.env.NEXT_PUBLIC_GRAPHQL_TOKEN}`,
      },
      body: JSON.stringify({
        query,
        variables,
      }),
    });

    const json = await res.json();

    if (json.errors) {
      console.error(json.errors);
      throw new Error(json.errors);
    }

    return json.data;
  } catch (error) {
    console.error(error);
    throw new Error(error);
  }
};
```

Am I doing something obviously wrong here?

**Comment by howells** [comment]
Browsing various issue threads makes me wonder whether the culprit is `<Link />` components prefetching the targets, so with every page load, calls are being made to the target routes endlessly, irregardless if they're visited. Would disabling prefetch help with this issue?

**Reply by barberdt** [reply]
@howells This is exactly what is happening as far as I can tell. See https://github.com/vercel/next.js/discussions/20521 for somebody else also observing the same problem. I see below that you eventually turned off revalidation entirely, which is a huge bummer. Is that still your solution or have you reasoned any sort of workaround?

(Note in that discussion that I also pinged the Discord server on this. Will post back here if I get a reply.)

**Comment by howells** [comment]
Turned off revalidation across all pages to test. This chart shows going from an eye-watering 16,000 PHP requests to my API, to 0, all with just about 10 visitors an hour (!). There's some major issue with revalidation, unless I've got completely the wrong end of the stick.

**Reply by AlexMNet** [reply]
Hi! Did you ever figure this issue out? I am having the same trouble. Revalidate seems to work in mysterious ways. I am using Next 13.4.4 but not using the new App Router.
